closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java

```diff
@@ -88,19 +90,41 @@
     private ChaChaEngine engine;
 
     private static NativeCrypto nativeCrypto;
+    private static final Cleaner contextCleaner;
     private long context;
 
     private final ByteArrayOutputStream aadBuf;
 
     static {
         nativeCrypto = NativeCrypto.getNativeCrypto();
+        contextCleaner = CleanerFactory.cleaner();
+    }
+
+    private static final class ChaCha20CleanerRunnable implements Runnable {
+        private final long context;
+
+        public ChaCha20CleanerRunnable(long context) {
+            this.context = context;
+        }
+
+        @Override
+        public void run() {
+            /*
+             * Release the ChaCha20 context.
+             */
+            long ret = nativeCrypto.DestroyContext(context);
+            if (ret == -1) {
+                throw new ProviderException("Error in Native CipherBlockChaining");
+            }
```

Comment:
Please explain these differences:
* this no longer checks that `context != -1`
* this didn't use to throw `ProviderException`.

Reply:
added error code check and throw exception if error is returned. the context != -1 check now happens in the constructor per your suggestion below.

Reply:
Throwing `ProviderException` here is appropriate: it would simply be ignored.

Reply:
are there any further updates to be made?

Reply:
There's no point in throwing an exception that will be simply ignored.